modules/message/src/main/JsonView.scala

```diff
@@ -33,9 +38,41 @@ final class JsonView() {
 
   def threadPost(thread: Thread, post: Post): JsValue =
     Json.obj(
-      "sender" -> thread.senderOf(post),
-      "receiver" -> thread.receiverOf(post),
+      "sender" -> user(thread.senderOf(post)),
+      "receiver" -> user(thread.receiverOf(post)),
       "text" -> post.text,
       "createdAt" -> post.createdAt
     )
+
+  def user(userId: String): JsValue =
+    Json.obj(
+      "username" -> userId,
+      "online" -> isOnline(userId)
+    )
+
+  private val errNames = Map(
+    "error.required" -> "is required",
+    "error.minLength" -> "too short",
+    "error.maxLength" -> "too long",
+    "Unknown username" -> "unknown")
+
+  def createThreadError(form: Form[_]): JsValue =
+    Json.obj (
+      "err" -> "Malformed request",
+      "errors" -> createThreadError(form.errors)
+    )
+
+  def createThreadError(errors: Seq[FormError]): JsValue = {
+    return errors.foldLeft(JsObject(Seq()))(accErrorMessages);
+  }
+
+  def accErrorMessages (acc: JsObject, e: FormError): JsObject = {
+    val result = acc \ e.key;
+
+    result match {
+      case JsDefined(v: JsArray) => acc + (e.key -> (v :+ JsString(errNames.getOrElse(e.message, e.message))))
+      case undefined: JsUndefined => acc + (e.key -> JsArray(Seq(JsString(errNames.getOrElse(e.message, e.message)))))
+    }
+  }
+
```

Comment:
not ok about the form stuff. That looks overly complicated, unrelated to messaging, and likely to reinvent the wheel.
I'll merge it anyway then try to simplify it.